### DISCUSSION TITLE
Reconstructing financial statement tables.

### DISCUSSION THREAD

**Discussion by elliotdes** [manusimidt/py-xbrl#123] [Q&A]
Would be a really useful feature if the parser was able to reconstruct financial statement tables (income statement, cash flow, balance sheet).

Is this currently possible?

**Comment by manusimidt** [comment]
Hey, 
hard to say, really depends also on the data. 
Generally, there is the concept of a presentation linkbase in XBRL. The presentation linkbase is responsible for encoding how different tables e.t.c are build up with the different facts, or in other words; how the data is presented in the Instance document. 

However, not all filings have a presentation linkbase associated to them. `py-xbrl` is able to parse the different linkbases, but currently offers no nice functionality to automatically combine and restructure the facts in the instance document according to a linkbase.

**Reply by elliotdes** [reply]
Thanks for the reply!

Do you have any examples of parsing the presentation linkbase into its distinctive tables? The example would have to be from a filing that has a presentation linkbase associated with it

It would be useful to know how to link items in the presentation linkbase to respective facts
